addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -59,17 +63,21 @@ options.registry.SocialMedia = options.Class.extend({
         // Handle element deletation.
         const entriesIds = entries.map(entry => entry.id);
         const anchorsEls = this.$target[0].querySelectorAll(':scope > a');
-        const deletedEl = Array.from(anchorsEls).find(aEl => !entriesIds.includes(aEl.id));
+        const deletedEl = Array.from(anchorsEls)
+            .find(aEl => !entriesIds.includes(this.links.find(linkEl => linkEl.el === aEl).id));
         if (deletedEl) {
             deletedEl.remove();
         }
         for (const entry of entries) {
-            let anchorEl = this.$target[0].querySelector(`#${entry.id}`);
-            if (!anchorEl) {
+            let anchorEl;
+            let link = this.links.find(linkEl => linkEl.id === entry.id);
+            if (!link) {
```

Comment:
You don't need:
- the extra variable
- the extra else condition
- the change of variable

```js
let {el:anchorEl} = this.links.find(linkEl => linkEl.id === entry.id) || {};
```
(Didn't test but something like that)